third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -622,16 +620,18 @@ void StreamPipeliner::scheduleDistanceOneDependencies() {
   }
 }
 
-void StreamPipeliner::scheduleRemainingToLastStage() {
+LogicalResult StreamPipeliner::scheduleRemainingToLastStage() {
   int lastStage = numStages - 1;
   // Assign the rest of the ops to the last stage.
   // Take care of the ordering of the ops - uses cannot be scheduled to the
   // cluster before the definition.
+  auto cluster = config[SCHED_TAIL].cluster;
   DenseMap<Operation *, tt::CoarseSchedule::Cluster> opToCluster;
   for (auto &op : forOp.getBody()->without_terminator()) {
     if (schedule.count(&op) == 0) {
-      auto schedType = isa<gpu::BarrierOp>(op) ? SCHED_COMPUTE : SCHED_TAIL;
-      opToCluster[&op] = config[schedType].cluster;
+      if (isa<gpu::BarrierOp>(op))
```

Comment:
Would it be better to make a pre-condition for the transformation out of it, so you wouldn't change anything (like creating stream ops, etc) if the loop is not going to be pipelined?